### ISSUE TITLE
API - potential memory leaks

### ISSUE THREAD

**Issue by oopsmonk** [iotaledger/iota.c#76]
## Bug description

```c
iota_str_t *cmd = iota_str_new(conf->url);
if (cmd == NULL) {
 ...
  return -1;
}

if (iota_str_append(cmd, cmd_info)) {
  ...
  return -1;
}
```

The snippet could cause memory leaks at runtime. if `iota_str_append` returns an error.
this scenario could be found in RESTful API implementations.
- [x] get_health
- [x] get_message_by_id
- [ ] get_node_info
- [x] get_output
- [x] get_outputs_from_address
- [x] get_tips

**Comment by bernardoaraujor** [comment]
I'm solving on `get_node_info`

**Comment by oopsmonk** [comment]
`get_health`, `get_message_by_id`, `get_output`, `get_outputs_from_address`, `get_tips` should be fixed in #81

**Comment by oopsmonk** [comment]
reopened this issue, it's blocked by #78
